library/src/commonMain/kotlin/com/badoo/reaktive/observable/Zip.kt

```diff
@@ -0,0 +1,231 @@
+package com.badoo.reaktive.observable
+
+import com.badoo.reaktive.disposable.CompositeDisposable
+import com.badoo.reaktive.disposable.Disposable
+import com.badoo.reaktive.utils.lock.newLock
+import com.badoo.reaktive.utils.lock.synchronized
+
+fun <T, R> Collection<Observable<T>>.zip(mapper: (List<T>) -> R): Observable<R> =
+    observableByEmitter { emitter ->
+        val disposables = CompositeDisposable()
+        emitter.setDisposable(disposables)
+        val lock = newLock()
+        val values = List<MutableList<T>>(size) { mutableListOf() }
```

Comment:
LinkedList instead of mutableListOf() as we always remove from the head.

Reply:
There is no LinkedList nor any queue in Kotlin native, I will try to write simple queue

Reply:
Ah, hadn't thought of that. Perhaps we could use an expect or something?